src/platform/mbed/ConnectivityManagerImpl.cpp

```diff
@@ -352,6 +352,12 @@ CHIP_ERROR ConnectivityManagerImpl::OnStationConnecting()
 {
     ChipLogProgress(DeviceLayer, "Event - StationConnecting");
 
+    // Update WiFi station state and propagate it if necessary
+    if (mWiFiStationState == kWiFiStationState_Connected)
+    {
+        return CHIP_NO_ERROR;
+    }
+
```

Comment:
What is the reason for this change?

Reply:
The Mbed stack can report this event while being connected. I want to prevent a connecting event to be propagated to the chip stack if we are already connected.

Reply:
Is this situation does not mean that we have lost the connection?

Reply:
No it doesn't we will receive a different message for disconnection 😓 .

Reply:
Okay, so the condition below won't be needed, it's redundant